pkg/network/modules.go

```diff
@@ -169,3 +169,9 @@ func (m *Module) SetIGateMap(gMap map[Gate]PipelineModule) {
 func (m *Module) SetEGateMap(gMap map[Gate]PipelineModule) {
 	m.IGates = gMap
 }
+
+func MakeGateMap() map[Gate]PipelineModule {
+    rGates := make(map[Gate]PipelineModule)
+    rGates[0] = nil
```

Comment:
will this work for ingress gates with bess? I know a lot bess modules only have a single ingress gate (0) for their module.

Reply:
It can be zero actually. There are modules with no iGates.
Ditto for eGates.
Bess gets around this by having two constants for each module which are fed into the registration macro.

Reply:
So the right way of doing it should be having MaxIGates and MaxEGates as part of the struct and building the map accordingly.

Reply:
The only place where IGates are used are replicator and forwarder. 
In the replicator gate 0 is immediately set to the forwarder after being added in the creation function.
In the l2forward case it should not work anyway as it is single ingress gate in Bess +/- my patches and the go code assumes it is multiple - I will have to fix that to get a pipeline per port as intended.

Reply:
We need to be careful about separating bess specific stuff from abstract nimbess stuff. What I mean is, just because a BESS l2 fwd module only allows 1 ingress gate, doesn't necessarily mean a nimbess switch should. I think it is something we should discuss in the future.

Reply:
The Ingress and Egress Port inherit module and have igates,egates. The egress port will connect its next available ingress gate to the last module of the pipeline. Just need to test if this works if it is non-zero with your changes. I'll try it out.

Reply:
I think I fixed the NPE, can you retest please?